contracts/FlyoverDiscovery.sol

```diff
@@ -0,0 +1,266 @@
+        return _isRegisteredForPegout(addr) && _collateralManagement.getPegOutCollateral(addr) >= _minCollateral;
+    }
+
+    // ------------------------------------------------------------
+    // Getter Functions
+    // ------------------------------------------------------------
+
+    /// @notice Returns the current minimum collateral threshold per side
+    /// @return minCollateral The minimum collateral value
+    function getMinCollateral() external view returns (uint256) {
+        return _minCollateral;
+    }
+
+    /// @notice Returns the last assigned provider id
+    /// @return lastId Last provider id
+    function getProvidersId() external view returns (uint) {
+        return lastProviderId;
+    }
+
+    /// @notice Returns the resignation starting block for an LP
+    /// @param addr The LP address
+    /// @return resignationBlock Block number when resignation started, or 0 if not resigned
+    function getResignationBlockNum(address addr) external view returns (uint256) {
+        return _resignationBlockNum[addr];
+    }
+
+    // ------------------------------------------------------------
+    // FlyoverDiscovery Private Functions
+    // ------------------------------------------------------------
+
+    function _shouldBeListed(Flyover.LiquidityProvider storage lp) private view returns(bool){
+        return (_isRegistered(lp.providerAddress) || _isRegisteredForPegout(lp.providerAddress)) && lp.status;
+    }
+
+    function _validateRegistration(
+        string memory name,
+        string memory apiBaseUrl,
+        Flyover.ProviderType providerType,
+        address providerAddress
+    ) private view {
+        if (providerAddress != msg.sender || providerAddress.code.length != 0) revert NotEOA(providerAddress);
+
+        if (
+            bytes(name).length < 1 ||
+            bytes(apiBaseUrl).length < 1
+        ) {
+            revert InvalidProviderData(name, apiBaseUrl);
+        }
+
+        if (providerType > type(Flyover.ProviderType).max) revert InvalidProviderType(providerType);
+
+        if (_resignationBlockNum[providerAddress] != 0) {
+            revert AlreadyRegistered(providerAddress);
+        }
+
+        // Check minimum collateral requirement
+        uint256 requiredCollateral = _minCollateral;
+        if (providerType == Flyover.ProviderType.Both) {
+            requiredCollateral = _minCollateral * 2;
+        }
+        if (msg.value < requiredCollateral) revert InsufficientCollateral(msg.value);
+    }
+
+    function _getProvider(address providerAddress) private view returns (Flyover.LiquidityProvider memory) {
+        for (uint i = 1; i <= lastProviderId; ++i) {
+            if (_liquidityProviders[i].providerAddress == providerAddress) {
+                return _liquidityProviders[i];
+            }
+        }
+        revert Flyover.ProviderNotRegistered(providerAddress);
+    }
+
+    function _isRegistered(address addr) private view returns (bool) {
```

Comment:
since the resignation part is being moved to CollateralManagement then you can remove this one and the one below and call the other contract to check